app/models/miq_retire_task.rb

```diff
@@ -0,0 +1,90 @@
+class MiqRetireTask < MiqRequestTask
+  validate :validate_request_type, :validate_state
+
+  AUTOMATE_DRIVES = true
+
+  def self.get_description(req_obj)
+    name = nil
+    if req_obj.source.nil?
```

Comment:
`name = if req_obj.source.nil?` then you can get rid of all of the other `name = `inside the conditional